test/Microsoft.Dnx.Testing.CommonUtils/Directory/Dir.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Dnx.Testing
+{
+    public class Dir
+    {
+        public const string EmptyFile = "";
+
+        private readonly Dictionary<string, object> _nodes = new Dictionary<string, object>();
+        private Func<FileInfo, object> _readFile;
+
+        public Dir(string rootPath)
+            : this()
+        {
+            if (!Directory.Exists(rootPath))
+            {
+                throw new DirectoryNotFoundException();
```

Comment:
It would be more proper to throw `new ArgumentException("Directory was not found", nameof(rootPath))`